security-privacy.html

```diff
@@ -207,90 +211,161 @@
-        The preferred type of this NDEF record is NFC Forum External Type
-        (TNF=0x04), with the External Type field set to e.g.
-        <code>urn:nfc:ext:w3.org:web-nfc-wl-write</code>.
-        The payload is a string list of URL's, delimited by white space.
-      </p>
-    </section>
-    <section> <h3>Web NFC black list for reading</h3>
-      <p>
-        Web NFC tags MAY also contain another special NDEF record containing a
-        list of URL's specifically NOT permitted to read to the tag.
-        The preferred type of this NDEF record is NFC Forum External Type
-        (TNF=0x04), with the External Type field set to e.g.
-        <code>urn:nfc:ext:w3.org:web-nfc-bl-read</code>.
-        The payload is a string list of URL's, delimited by white space.
-      </p>
-    </section>
-    <section> <h3>Web NFC black list for writing</h3>
-      <p>
-        Web NFC tags MAY also contain another special NDEF record containing a
-        list of URL's specifically NOT permitted to write to the tag.
-        The preferred type of this NDEF record is NFC Forum External Type
-        (TNF=0x04), with the External Type field set to e.g.
-        <code>urn:nfc:ext:w3.org:web-nfc-bl-write</code>.
-        The payload is a string list of URL's, delimited by white space.
-      </p>
-    </section>
+  </section>
+  <section> <h3>Web NFC using encoded metadata</h3>
+    <p>
+      Web NFC may also define an own NDEF extended record format,
+      where the preferred type of this NDEF record is NFC Forum External Type
+      (TNF=0x04), with the External Type field set to
+      <code>urn:nfc:ext:w3.org:webnfc</code>. The payload type is
+      <code>application/webnfc+json</code> (encoded as UTF-8 by default). All
+      data is JSON, with the following schema:
+      <ul>
+        <li>
+          <code>id</code>: <code>"string"</code>; contains the Web NFC
+          identifier, e.g. the SHA256 signature of <code>"w3.org/web-nfc”</code>
+        </li>
+        <li>
+          <code>scope</code>: <code>"string"</code>; contains the origin of the
+          data; it can be also stored at the NDEF record's
+          <code>id</code> field.
+        </li>
+        <li>
+          <code>messageType</code>: <code>"string"</code>; contains the type of
+          the data, either <code>"text"</code>, <code>"url"</code>,
+          <code>"json"</code>, or <code>"blob"</code>.
+        </li>
+        <li>
+          <code>message</code>: <code>"string"</code>; contains the payload data
+          encoded as text (text, URL and JSON), and for blob encoded as
+          [[Base64]].
+        </li>
+        <li>
+          <code>access</code>: <code>"array"</code>; contains tuples
+          for the desired access type (read-write, or read-only),
+          and the URL scopes that are explicitly allowed to access the data,
+          for instance
+          <code>{ "scope": "https://mydomain.com", "write": "true"}</code>.
```

Comment:
We really need one definition of scopes as it is also used by Service Workers @PaulKinlan @jakearchibald